stablehlo/dialect/TypeInference.cpp

```diff
@@ -4309,12 +4223,11 @@ LogicalResult verifySelectAndScatterOp(
   auto selectResultType = selectResult[0].getType().dyn_cast<ShapedType>();
   // select_and_scatter_c9
   if (!selectResultType || !selectResultType.getElementType().isInteger(1) ||
-      (selectResultType.hasRank() &&
-       selectResultType.cast<RankedTensorType>().getRank() != 0))
-    return emitOptionalError(
-        location,
-        "expects the return-type of select-region to be tensor<i1>, but got: ",
-        selectResult[0].getType());
+      selectResultType.cast<RankedTensorType>().getRank() != 0)
```

Comment:
Do we need the cast here?

Reply:
Done. Also I cleaned up the declaration of `selectResultType` since it's right there.